app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -22,6 +30,39 @@ def vm_size
       @azure_vm.hardware_profile.vm_size
     end
 
+    def platform
+      @azure_vm.storage_profile.os_disk.os_type
+    end
+
+    def username
+      @azure_vm.os_profile.admin_username
+    end
+
+    def password
+      @azure_vm.os_profile.admin_password
+    end
+
+    def ssh_key_data
+      # since you can only give one additional
+      # sshkey through foreman's UI
+      sshkey = @azure_vm.os_profile.linux_configuration.ssh.public_keys[1]
+      return unless sshkey.present?
+      sshkey.key_data
+    end
+
+    def premium_os_disk
```

Comment:
Same like `platform`, can't you set value for premium_od_disk using @azure_vm.storage_profile.os_disk.managed_disk?

Would it be possible to fetch `storage_account_type` value from  `managed_disk`?